Feature/#12 manager routing

guglieLLM opened the pull request:
# Feature/Manager routing

   - Created lazy module manager
   - Added modules with relative components :
    - landlord :
        - list
        - new
        - edit
        - view

    - tenant :
        - list
        - view
        - edit

    - reservation :
        - list
        - details

   - Configured routes
   - Fixed style and lint

# Developer Checklist

- [ ] Updated documentation or README.md
- [ ] If adding new feature(s), added and ran unit tests
- [ ] If create new release, bumped version number
- [X] Ran `npm run style:fix` for code style enforcement
- [X] Ran `npm run lint:fix` for linting
- [ ] Ran `npm audit` to discover vulnerabilities

CosNig committed:
Add manager module with lazy loading and container component

Gabriele-bil committed:
Add reservation module with lazy loading

CarmeloB4 committed:
create lazy loading tenant module

Gabriele-bil committed:
Add list component and configured reservation module

CarmeloB4 committed:
create list component

CarmeloB4 committed:
create route path for list component

DaveCode87 committed:
add details component and configured reservation module

DaveCode87 committed:
fix style and lint

CosNig committed:
create edit component

CosNig committed:
create view component

CosNig committed:
create route paths for edit and view components

CosNig committed:
fix style

Matteo-Oliva committed:
Add module landlord.

Matteo-Oliva committed:
Configured manager routing with lazy loading.

Matteo-Oliva committed:
Add component list.

Matteo-Oliva committed:
Add component new.

guglieLLM committed:
Add edit component

guglieLLM committed:
Add view component

Matteo-Oliva committed:
Fix style and lint.

CarmeloB4 committed:
Update manager-routing.module.ts

guglieLLM committed:
Merge pull request #39 from pc-codingavision/feature/manager/#14-tenant

Feature/manager/#14 tenant

guglieLLM committed:
Merge branch 'feature/#12-manager_routing' into feature/manager/#13-landlord

guglieLLM committed:
Merge pull request #40 from pc-codingavision/feature/manager/#13-landlord

Feature/manager/#13 landlord

guglieLLM committed:
Merge branch 'feature/#12-manager_routing' into feature/manager/#15-reservation

guglieLLM committed:
Merge pull request #38 from pc-codingavision/feature/manager/#15-reservation

Feature/manager/#15 reservation

Gabriele-bil committed:
fix style and lint

pc-codingavision commented on src/app/modules/core/manager/manager-routing.module.ts:
```diff
@@ -0,0 +1,22 @@
+import { NgModule } from '@angular/core'
+import { RouterModule, Routes } from '@angular/router'
+
+import { ManagerComponent } from './container/manager-container.component'
```
Da preferire `ManagerContainerComponent` come nome del componente.

pc-codingavision commented on src/app/modules/core/manager/manager.module.ts:
```diff
@@ -0,0 +1,20 @@
+  imports: [
+    CommonModule,
+    ManagerRoutingModule,
+    LandlordModule,
```
Se un modulo viene definito lazy, non va importato nel parent module.

pc-codingavision replied:
Rimuovere

pc-codingavision commented on src/app/modules/core/manager/manager.module.ts:
```diff
@@ -0,0 +1,20 @@
+    CommonModule,
+    ManagerRoutingModule,
+    LandlordModule,
+    TenantModule,
```
Se un modulo viene definito lazy, non va importato nel parent module.

pc-codingavision replied:
Rimuovere

pc-codingavision commented on src/app/modules/core/manager/manager.module.ts:
```diff
@@ -0,0 +1,20 @@
+    ManagerRoutingModule,
+    LandlordModule,
+    TenantModule,
+    ReservationModule,
```
Se un modulo viene definito lazy, non va importato nel parent module.

pc-codingavision replied:
Rimuovere

pc-codingavision commented:
Il reservation module non ha un container component.

pc-codingavision commented:
Il landlord module non ha un container component.

pc-codingavision commented:
Il tenant module non ha un container component.

Gabriele-bil committed:
Add containers component to each module

Gabriele-bil committed:
Rename container folder to manager-container

Gabriele-bil committed:
Fix import directories path

Gabriele-bil committed:
Modify Landlord selector name
Modify Landlord component name

Gabriele-bil committed:
Modify Reservation selector name
Modify Reservation component name

Gabriele-bil committed:
Modify Tenant selector name
Modify Tenant component name

Gabriele-bil committed:
Fix import for ManagerReservationListComponent

guglieLLM commented:
Add containers component to each module,
Modify selectors name
Modify components name

pc-codingavision commented on src/app/modules/core/manager/manager-container/manager-container.component.ts:
```diff
@@ -0,0 +1,12 @@
+  templateUrl: './manager-container.component.html',
+  styleUrls: ['./manager-container.component.scss'],
+})
+export class ManagerComponent implements OnInit {
```
Change the component name to reflect the file name.

CosNig committed:
Prova

pc-codingavision committed:
Merge pull request #53 from pc-codingavision/test

Prova

CosNig committed:
Provs2

pc-codingavision committed:
Merge pull request #54 from pc-codingavision/test

Provs2

CosNig committed:
Terza prova

pc-codingavision committed:
Merge pull request #55 from pc-codingavision/test

Terza prova

Matteo-Oliva committed:
Changed container folders names in container
Add components folder and move inside relative components
Removed wrong imports from manager-module

Matteo-Oliva committed:
Fix style

Matteo-Oliva committed:
Merge branch 'develop' into feature/#12-manager_routing

pc-codingavision commented on src/app/modules/core/manager/container/manager-container.component.html:
```diff
@@ -0,0 +1 @@
+<p>manager works!</p>
```
The container folder should go inside a `components` folder.

pc-codingavision commented on src/app/modules/core/manager/container/manager-container.component.spec.ts:
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed, async } from '@angular/core/testing'
```
The container folder should go inside a `components` folder.

pc-codingavision commented on src/app/modules/core/manager/container/manager-container.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
The container folder should go inside a `components` folder.

pc-codingavision commented on src/app/modules/core/manager/landlord/components/edit/manager-landlord-edit.component.html:
```diff
@@ -0,0 +1 @@
+<p>edit works!</p>
```
Please, rename the file, the component class, all the related files and the selector in `landlord-edit` instead of `manager-landlord-edit`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/landlord/components/list/manager-landlord-list.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `landlord-list` instead of `manager-landlord-list`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/landlord/components/new/manager-landlord-new.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `landlord-new` instead of `manager-landlord-new`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/landlord/components/view/manager-landlord-view.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `landlord-view` instead of `manager-landlord-new`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/landlord/container/landlord-container.component.html:
```diff
@@ -0,0 +1 @@
+<p>landlord-container works!</p>
```
Please move the `container` folder inside a `components` folder.

pc-codingavision commented on src/app/modules/core/manager/reservation/components/details/manager-reservation-details.component.html:
```diff
@@ -0,0 +1 @@
+<p>details works!</p>
```
Please, rename the file, the component class, all the related files and the selector in `reservation-details` instead of `manager-reservation-details`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/reservation/components/details/manager-reservation-details.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `reservation-details` instead of `manager-reservation-details`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/reservation/components/list/manager-reservation-list.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `reservation-list` instead of `manager-reservation-list`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/reservation/container/reservation-container.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please move the `container` folder inside a `components` folder. Thanks.

pc-codingavision commented on src/app/modules/core/manager/tenant/components/edit/manager-tenant-edit.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `tenant-details` instead of `manager-tenant-details`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/tenant/components/list/manager-tenant-list.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `tenant-list` instead of `manager-tenant-list`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/tenant/components/view/manager-tenant-view.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
```
Please, rename the file, the component class, all the related files and the selector in `tenant-view` instead of `manager-tenant-view`. Most probably this component will be shared with the landlord module. Thanks.

pc-codingavision commented on src/app/modules/core/manager/tenant/container/tenant-container.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core'
+
```
Please move the `container` folder inside a `components` folder. Thanks

pc-codingavision commented on src/app/modules/core/manager/tenant/tenant-routing.module.ts:
```diff
@@ -0,0 +1,20 @@
+import { TenantContainerComponent } from './container/tenant-container.component'
+
+const routes: Routes = [
+  { path: 'container', component: TenantContainerComponent },
```
This should be mapped to the empty path inside that `container`. Thanks.

pc-codingavision commented on src/app/modules/core/package.info:
```diff
@@ -1 +1,4 @@
 Package for the core modules
+ciao a tutti
```
@CosNig please, remove those stuff we added for our tests. Thanks.

CosNig committed:
Removed lines for test purpose, moved container component into components folder, renamed components name and related files, add empty path for container components

CosNig committed:
Resolved merge conflicts

Gabriele-bil committed:
Merge branch 'develop' into feature/#12-manager_routing
